File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.js
var maxScore = function(cardPoints, k) {
    let total = cardPoints.reduce((a, b)=> a + b, 0)
    let windowSize = cardPoints.length - k
    let windowSum = 0
    minWindowSum = Infinity
    let output = 0

    for (let i = 0; i < windowSize; i++) {
        windowSum += cardPoints[i]
    }

    output = total - windowSum

    for (let i = windowSize; i < cardPoints.length; i++) {
        windowSum += cardPoints[i] - cardPoints[i - windowSize]
        minWindowSum = Math.min(windowSum, minWindowSum)
    }
    
    return Math.max(total - minWindowSum, output)

};

File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        total_score = sum(cardPoints)
        window_size = len(cardPoints) - k
        window_score = sum(cardPoints[:window_size])
        # current_score = total_score - window_score
        output = total_score - window_score

        for i in range(window_size, len(cardPoints)):
            window_score += cardPoints[i] - cardPoints[i - window_size]
            output = max(output, total_score - window_score)

        return output


        